src/System.Buffers/ref/System.Buffers.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// ------------------------------------------------------------------------------
+// Changes to this file must follow the http://aka.ms/api-review process.
+// ------------------------------------------------------------------------------
+
+namespace System.Buffers
+{
+    public abstract class ArrayPool<T>
+    {
+        public const int DefaultNumberOfArraysPerBucket = 50;
+        public const int DefaultArrayLength = 1024000;
+
+        public static ArrayPool<T> Shared{ get; }
+        public static ArrayPool<T> Create(int maxArrayLength = DefaultArrayLength, int arraysPerBucket = DefaultNumberOfArraysPerBucket);
+        public abstract int RentedBufferCount { get; }
+        public abstract int TotalBufferCount { get; }
+        public abstract T[] Rent(int length);
+        public abstract T[] Enlarge(T[] original, int newLength);
+        public abstract void Return(T[] buffer, bool clearArray = false);
+    }
+
+    internal sealed class DefaultArrayPool<T> : ArrayPool<T>
```

Comment:
> Does this new library require a ref folder?

@ericstj, @weshaggard?  My understanding matched Justin's, that no ref folder is needed (I'd initially had one for System.Threading.Tasks.Extensions and removed it on recommendation from @nguerrera).

> If the ref assembly is required for this library, do internal types need to be listed?

No.